### PR TITLE
[MINT-2519] [FE] Notification: Data exchange approach is complete

### PR DESCRIPTION
**Pull request by garyjzhao**
# MINT-2519

## Description

* add `isDataExchangeApproach` typeguard for notification settings
* Update Notification Settings component to take in new Data Exchange Approach notification subscriptions
* Update `GetNotificationSttings` query and its type
* Update i18n file

## How to test this change

1.  Navigate to Notification Settings page and verify it matches figma
2. Subscribe to both notifications types
3. Navigate back to collaboration area and "complete" the Data Exchange form (just fast forward to last page and mark it as complete)
4. Ensure notification shows up on notification page and matches figma
5. Try the unsubscribe link and ensure it works as epected
`http://localhost:3005/notifications/settings?unsubscribe_email=DATA_EXCHANGE_APPROACH_MARKED_COMPLETE`

## PR Author Checklist

- [x] I have provided a detailed description of the changes in this PR.
- [x] I have provided clear instructions on how to test the changes in this PR.
- [ ] I have updated tests or written new tests as appropriate in this PR.
- [ ] Updated the [Postman Collection](../MINT.postman_collection.json) if necessary.

## PR Reviewer Guidelines

- It's best to pull the branch locally and test it, rather than just looking at the code online!
- When approving a PR, provide a reason _why_ you're approving it
  - e.g. "Approving because I tested it locally and all functionality works as expected"
  - e.g. "Approving because the change is simple and matches the Figma design"
- Don't be afraid to leave comments or ask questions, especially if you don't understand why something was done! (This is often a great time to suggest code comments or documentation updates)
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.

### CODE REVIEW

File: src/components/FormFooter/index.tsx [hunk 1]
```diff
@@ -3,7 +3,7 @@ import { useTranslation } from 'react-i18next';
 import { useHistory } from 'react-router-dom';
 import { Button, Icon } from '@trussworks/react-uswds';
 
-type FormFooterProps = {
+type SubmittionFooterProps = {
```
Comment (garyjzhao) [inline src/components/FormFooter/index.tsx:6]: This has nothing to do with my code, but making these changes to address type errors

File: src/features/ModelPlan/Collaborators/AddCollaborator/index.tsx [hunk 2]
```diff
@@ -357,8 +357,12 @@ const Collaborators = () => {
...
                             undefined,
                             isModelLead && isLastModelLead(allCollaborators)
-                              ? TeamRole.MODEL_LEAD
-                              : ''
+                              ? {
+                                  [TeamRole.MODEL_LEAD]: collaboratorsT(
+                                    `teamRoles.options.${TeamRole.MODEL_LEAD}`
+                                  )
+                                }
+                              : undefined
```
Comment (garyjzhao) [inline src/features/ModelPlan/Collaborators/AddCollaborator/index.tsx:365]: This has nothing to do with my code, but making these changes to address type errors

**Review by patrickseguraoddball — CHANGES_REQUESTED** [review]
I'm getting emails, even though I dont have those selected settings.  Not sure if I'm missing a step of the instructions. 

Screenshot 2024-10-29 at 2 39 11 PM



Screenshot 2024-10-29 at 2 39 41 PM

Also not getting any notifications in the UI when have the settings selected

**Comment by patrickseguraoddball** [comment]
Not seeing this notification come through

Screenshot 2024-10-29 at 2 55 00 PM

**Review by ClayBenson94 — APPROVED** [review]

**Pull request merged**